lib/fizzy/execute.cpp

```diff
@@ -388,6 +388,27 @@ execution_result execute(Instance& instance, FuncIdx func_idx, std::vector<uint6
                 stack.push(ret.stack[0]);
             break;
         }
+        case Instr::return_:
+        {
+            // TODO: Not needed, but satisfies the assert in the end of the main loop.
+            labels.clear();
+
+            assert(func_idx < instance.module.funcsec.size());
```

Comment:
This assert should go to the top of the function.

Reply:
I don't think it should. Functype is only needed for `return` instruction, no point requiring it for every execution.

Reply:
We should validate the number of input arguments based on the functype.

Reply:
Not sure what this is needed all the time, but probably should be discussed as a part of another PR.